src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -25,9 +24,25 @@ object SingleWorkflowRunnerActor {
  * Designed explicitly for the use case of the 'run' functionality in Main. This Actor will start a workflow,
  * print out the outputs when complete and then shut down the actor system. Note that multiple aspects of this
  * are sub-optimal for future use cases where one might want a single workflow being run.
+ *
+ * NOTE: This Actor listens to FSM transition messages to keep an eye on the WorkflowActor. However, when the FSM
+ * transitions to a "terminated" state, it is just starting to asynchronously begin more work in response to the same
+ * message. This work includes persisting the run state to the DB, before it sends a special message to itself to
+ * actually halt the Actor instance. (Side note, the halt only appears to be received in state WorkflowSucceeded, not
+ * WorkflowFailed?)
```

Comment:
Yeah this does sounds essentially like DSDEEPB-1631, we should not transition to a terminal state until we've persisted the status change in the db.  i.e. it's not cool to be spinning off the database update Future in the onTransition handler.  :frowning: